curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6153908a366afb4d57185c8d.md

```diff
@@ -7,16 +7,41 @@ dashedName: step-12
 
 # --description--
 
-Flexbox can be thought of as having two axes, the main axis and the cross axis. The main axis is determined by the `flex-direction` property. If `flex-direction` is set to `row` or `row-reverse`, the main axis is horizontal. If `flex-direction` is set to `column` or `column-reverse`, the main axis is vertical.
+When working with flexbox you should think of it as having two axis. The main axis and the cross axis. The main axis is defined by `flex-direction`, which has four possible values:
 
-Give your `#gallery` selector a `flex-direction` property set to `row`.
+- `row`
+- `row-reverse`
+- `column`
+- `column-reverse`
+
+When `flex-direction` is set to `row` or `row-reverse`, the main axis is the horizontal inline direction. When `flex-direction` is set to `column` or `column-reverse`, the main axis is the vertical block direction.
+
+Default value for `flex-direction` is `row` when making an element a flex container.
+
+---
+
+You are encouraged to try out the different values and see how they affect the layout. When you are done, set an explicit `flex-direction` of `row` on the `.gallery` element.
+
+Then give the `.gallery` element a `max-width` of `1400px` and center it using an auto `margin` by setting `0 auto` as the `margin` value.
```

Comment:
I assume you are on a phone. Not sure if being able to see the editable region and all the text at the same time is really necessary. The instructions are at the bottom and you should be able to see them and the editable region.

In my opinion, we can't prioritize screen real estate that much higher than the subject we are teaching. But I agree, that step does have a lot of text, it already did. I can see if I can compress the information down, but it might take some time. But I won't remove as much as you are suggesting as it doesn't teach what the property does at all.

I might have to move the max-width and auto margin to a different step as well.

Reply:
> I assume you are on a phone. Not sure if being able to see the editable region and all the text at the same time is really necessary. The instructions are at the bottom and you should be able to see them and the editable region.

It was the laptop 😅

> In my opinion, we can't prioritize screen real estate that much higher than the subject we are teaching. But I agree, that step does have a lot of text, it already did. I can see if I can compress the information down, but it might take some time. But I won't remove as much as you are suggesting as it doesn't teach what the property does at all.

Sure, I though we were introducing concepts, not going in depth like the legacy RWD.

Reply:
I do like the idea of making new step, so people can focus on new concepts.

Reply:
Should be a bit better now. I added the direction information to the value list and move the max-with and centering to step 16.